api/tests/integration/infrastructure/repositories/organization-repository_test.js

```diff
@@ -292,17 +291,21 @@ describe('Integration | Repository | Organization', function() {
 
     it('should return the organizations that matches the filters', async () => {
       // given
-      const filters = { type: 'PRO' };
+      const externalIds = ['1234567', '1234569'];
 
       // when
-      const actualOrganizations = await organizationRepository.findBy(filters);
+      const foundOrganizations = await organizationRepository.findByExternalIds(externalIds);
 
       // then
-      expect(actualOrganizations).to.have.lengthOf(1);
-      expect(actualOrganizations[0]).to.be.an.instanceof(Organization);
-      expect(actualOrganizations[0].code).to.equal(organizations[0].code);
-      expect(actualOrganizations[0].type).to.equal('PRO');
-      expect(actualOrganizations[0].type).to.equal(organizations[0].type);
+      expect(foundOrganizations).to.have.lengthOf(2);
+      expect(foundOrganizations[0]).to.be.an.instanceof(Organization);
+      expect(foundOrganizations[0].externalId).to.equal(organizations[0].externalId);
+      expect(foundOrganizations[0].id).to.equal(organizations[0].id);
+      expect(foundOrganizations[0].type).to.be.undefined;
+      expect(foundOrganizations[1]).to.be.an.instanceof(Organization);
+      expect(foundOrganizations[1].externalId).to.equal(organizations[2].externalId);
+      expect(foundOrganizations[1].id).to.equal(organizations[2].id);
+      expect(foundOrganizations[1].type).to.be.undefined;
```

Comment:
C'est assez dur à lire ce genre de gros block. Y'a beaucoup de bruit pour pas beaucoup d'informations. Généralement y'a deux solutions pour contourner le problème que j'aime bien perso.

- Extraire ça dans des `it` séparé pour marquer les intentions différentes. Je testerai pas le `instanceof(Organization)` ça alourdit les test, généralement on test la conversion juste dans le get dans certains repo (convention que j'aime bien)
- Utiliser des variables avec la destructuration pour utiliser `deep.members` quand on récupère des clés en trop

La solution 1 ça donnerait genre : 
```
it('should only retrieve id and external id', ...) 
it('should only retrieve matching organizations') 
```

La solution 2 : 

ça consiste à mettre dans une variable le résultat des appels au `databaseBuilder` pour avoir les même objets en sortie de répo qu'à l'insertion, et d'utiliser `deep.members` pour s'affranchir de l'ordre, comme dans ce fichier : https://github.com/1024pix/pix/blob/dev/api/tests/integration/infrastructure/repositories/knowledge-element-repository_test.js

Reply:
Fait ! Bon commentaire ;)